packages/@core.next/auth.jsx

```diff
@@ -69,11 +69,17 @@ const AuthProvider = ({ children, initialUserValue }) => {
 
     const { data: userData, loading: userLoading, error: userError, refetch } = useQuery(USER_QUERY)
 
+    const [isUserLoading, setIsUserLoading] = useState(userLoading)
+
     useEffect(() => {
         if (userData) onData(userData)
         if (userError) onError(userError)
     }, [userData, userError])
 
+    useEffect(() => {
+        setIsUserLoading(userLoading)
```

Comment:
it's better to check the value before the call setState to avoid redundant rerendering. But, in that case it's not required.

Reply:
if the `userLoading` variable is the same as current state, re-render won't occurred.
https://reactjs.org/docs/hooks-reference.html#bailing-out-of-a-state-update